src/test/ui/issue-23302-3.stderr

```diff
@@ -1,30 +1,25 @@
-error[E0391]: cyclic dependency detected
-  --> $DIR/issue-23302-3.rs:13:16
-   |
-LL | const B: i32 = A; //~ ERROR cyclic dependency detected
-   |                ^ cyclic reference
-   |
-note: the cycle begins when const checking if rvalue is promotable to static `A`...
+error[E0391]: cycle detected when const checking if rvalue is promotable to static `A`
```

Comment:
This is a bit jargon heavy. @nikomatsakis what would you recommend? I'm thinking something along the lines of

```
error[E0391]: cycle detected when taking `A` as static
  --> $DIR/issue-23302-3.rs:11:1
   |
LL | const A: i32 = B; //~ ERROR cycle detected
   | ^^^^^^^^^^^^^^^^^
   |
note: ...which requires checking which parts of `A` could be static...
  --> $DIR/issue-23302-3.rs:11:16
   |
LL | const A: i32 = B; //~ ERROR cycle detected
   |                ^
note: ...which requires taking `B` as static...
  --> $DIR/issue-23302-3.rs:13:1
   |
LL | const B: i32 = A;
   | ^^^^^^^^^^^^^^^^^
note: ...which requires checking which parts of `B` could be static...
  --> $DIR/issue-23302-3.rs:13:16
   |
LL | const B: i32 = A;
   |                ^
   = note: ...which again requires taking `A` as static, completing the cycle
```

Reply:
I have no idea what `taking as static` or `being static` means.

Reply:
From

```
cycle detected when const checking if rvalue is promotable to static `A`
```

I understand

```
cycle detected when [checking during compilation] if [the value being assigned] is [able to be stored to `const A` in static memory instead of stack slots]
```

That is too long for an error message, IMO, and I might also be wrong. That being said, not everybody, and certainly no newcomers know what "const checking", "rvalue" and ["promotable to static"](https://github.com/rust-lang/rfcs/blob/master/text/1414-rvalue_static_promotion.md) mean.

@nrc, @eddyb, @steveklabnik would you have a suggestion on a message with less jargon for this error?

That being said, I wouldn't block this PR on this at all.